### ISSUE TITLE
[QUESTION] How to use a named docker volume

### ISSUE THREAD

**Issue by remygrandin** [Lissy93/dashy#599]
### Question

Hi,

I'm trying to deploy Dashy on my docker cluster, however for that I need to be able to use a named volume for persistent storage.
I saw in your documentation an exemple of docker compose but with a direct file mapping for persistence.

I tried this
```
  dashy:
    image: lissy93/dashy
    container_name: BattleStation_Dashy
    # Pass in your config file below, by specifying the path on your host machine
    volumes:
      - persistent-Dashy:/public/
    # Set any environmental variables
    environment:
      - NODE_ENV=production
    # Specify your user ID and group ID. You can find this by running `id -u` and `id -g`
      - UID=500
      - GID=500
    # Specify restart policy
    restart: unless-stopped
    # Configure healthchecks
    healthcheck:
      test: ['CMD', 'node', '/app/services/healthcheck']
      interval: 1m30s
      timeout: 10s
      retries: 3
      start_period: 40s
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.battleStation_dashy_web.entrypoints=web"
      - "traefik.http.routers.battleStation_dashy_web.rule=Host(`XXXXXXXXX`)"
      - "traefik.http.routers.battleStation_dashy_websecure.entrypoints=websecure"
      - "traefik.http.routers.battleStation_dashy_websecure.tls=true"
      - "traefik.http.routers.battleStation_dashy_websecure.rule=Host(`XXXXXXXX`)"
```
Is there a way to specify the path of the config storage ? maybe with and ENV variable ?

PS : I reposted the question as I did't see a way to reopen the previous one auto closed by your bot :-)

### Category

Setup and Deployment

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [question hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by Lissy93** [comment]
So you should be able to create a volume, with a name (`--name` flag), then reference that volume within your compose, just by it's name (instead of path). And what you've done looks correct to me.

However, unless your volume contains the contents of the [`public/`](https://github.com/Lissy93/dashy/tree/master/public) directory, the app isn't going to run. Your best bet is to just pass in the `conf.yml` file as the volume, instead of the entire directory.
(I know that structure isn't ideal, I do plan on updating it sometime, but it would be a breaking change, so need to handle it carefully)

**Comment by XxAcielxX** [comment]
I also think we need a volume mount to `/app/public`,  as it holds all the configurations such as:
* conf.yml
* item-confs
* index.html (this needs to be editable to set the site's Title, it is by default set to `Dashy`)
* favicon.ico and other logos etc.

**Comment by remygrandin** [comment]
The main issue is, as far as I know it, named volume can only be mapped to a folder an not to an individual file. This require also that the folder it map to in the container be empty.

In most non hyper-local environment, you can only define named volume (as the container is not on a defined host and therefor is not allowed to map to a real path) and the volume start empty (and is not meant to be accessible other than via container.

I see 2 possibles solutions :

1) Move config to /public/config/config.yml
 - Pro : 
   - Allow easy mapping of folder to volume
 - Con :
   - Config file need to be created by the server if empty / not existing
   - Need extra check by the app to not break compatibility (if /public/config.yml exist, use it, of not use /public/config/config.yml)

2) Bootstrap /public/ folder on launch if empty folder
 - Pro :
   - easier solution of the 2 to implement ?
 - Con :
   - Extra static files stored in volume instead of the container
   - Require extra bootscript for the container to check for empty folder and copy from build folder
   - Need to retarget build output folder to other template folder (/public-template/ for exemple)
   - Problematic for upgrade ?

As you said, the priority should be non breaking change :-)

I will try on my side to extend your container with a custom dockerfile, maybe I can play with symlink to retarget the configfile elsewhere, we'll see.

**Comment by remygrandin** [comment]
I just saw your latest commits, especially [this one](https://github.com/Lissy93/dashy/commit/32ca8996d23570528954540376052fab5754b2e2)

It seem indeed that targeting the right folder fix the issue. The bootstraping from a empty folder work fine by targetting /app/public

You can probably consider my previous comment null and void :-) (and I will close this issue now that the documenation is fixed)

Thanks for this great tool ^^

**Comment by Lissy93** [comment]
Ohhh, I'm sorry I should have noticed in your original question that the app was missing 🤦 

I'm not sure why this was ever removed from the docs in https://github.com/Lissy93/dashy/commit/40a56ceb583712418c541e3ef838f0de2de99882 by StevKast 🤔
But this was raised in #601, and I added it back in

Feel free to reach out if you need anything else :)

**Comment by StevKast** [comment]
Yeah, I goofed and had a misunderstanding when I was troubleshooting on my own cluster. I got mixed up about how this containers dockerfile is not defining a volume vs how a lot of the linuxserver.io volumes. For example the last line of this file: [https://github.com/linuxserver/docker-sonarr/blob/master/Dockerfile](https://github.com/linuxserver/docker-sonarr/blob/master/Dockerfile).

Sorry for breaking things :( @Lissy93
